### PR TITLE
multi_asset_check decorator

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

**Review by erinkcochran87 — COMMENTED** [review]
It looks like there are API doc changes, but they're not showing up. Could you run `make apidoc-build` and then push the newly generated files up so I can check things out?

**Review by johannkm — COMMENTED** [review]
I think we should consider `@asset_multi_check`. I think there's the potential interpretation that this is a check for multi assets

internal thread: https://dagsterlabs.slack.com/archives/C047L6H0LF4/p1710791332784459

**Review by johannkm — APPROVED** [review]

File: python_modules/dagster/dagster/_core/definitions/decorators/asset_check_decorator.py [hunk 1]
```diff
@@ -237,6 +251,121 @@ def inner(fn: AssetCheckFunction) -> AssetChecksDefinition:
...
+    description: Optional[str] = None,
+    can_subset: bool = False,
+    compute_kind: Optional[str] = None,
+    op_tags: Optional[Mapping[str, Any]] = None,
+    resource_defs: Optional[Mapping[str, object]] = None,
+    required_resource_keys: Optional[Set[str]] = None,
+    retry_policy: Optional[RetryPolicy] = None,
+    config_schema: Optional[UserConfigSchema] = None,
+) -> Callable[[Callable[..., Any]], AssetChecksDefinition]:
+    """Defines a set of asset checks that can be executed together with the same function.
```
[the remark this answers is not in the document]
Comment (sryza) [reply python_modules/dagster/dagster/_core/definitions/decorators/asset_check_decorator.py:278]: 👍

**Comment by sryza** [comment]
> It looks like there are API doc changes, but they're not showing up. Could you run make apidoc-build and then push the newly generated files up so I can check things out?

@erinkcochran87  I have been banging my head against trying to get this to run locally, but alas to no avail. Would you strenuously object to merging it without this? FWIW I think the changes are pretty low risk.

**Comment by sryza** [comment]
Following up from https://github.com/dagster-io/dagster/pull/20528#issuecomment-2005119003, merging this because @dpeng817 needs it for a downstream PR, and didn't see a blocking review from you Erin. Will take a look at it in the published api docs and make sure it looks reasonable!

**Pull request merged**